src/main/java/org/kiwiproject/dropwizard/error/dao/jdk/JdbcApplicationErrorDao.java

```diff
@@ -0,0 +1,322 @@
+package org.kiwiproject.dropwizard.error.dao.jdk;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static org.kiwiproject.base.KiwiPreconditions.checkArgumentIsNull;
+import static org.kiwiproject.base.KiwiStrings.f;
+import static org.kiwiproject.collect.KiwiLists.first;
+import static org.kiwiproject.dropwizard.error.dao.ApplicationErrorDao.checkPagingArgumentsAndCalculateZeroBasedOffset;
+import static org.kiwiproject.dropwizard.error.dao.ApplicationErrorJdbc.nextOrThrow;
+import static org.kiwiproject.jdbc.KiwiJdbc.timestampFromZonedDateTime;
+
+import org.kiwiproject.dropwizard.error.dao.ApplicationErrorDao;
+import org.kiwiproject.dropwizard.error.dao.ApplicationErrorJdbc;
+import org.kiwiproject.dropwizard.error.dao.ApplicationErrorJdbc.ApplicationErrorJdbcException;
+import org.kiwiproject.dropwizard.error.dao.ApplicationErrorStatus;
+import org.kiwiproject.dropwizard.error.model.ApplicationError;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Implementation of {@link ApplicationErrorDao} that uses plain JDBC, and therefore does not require
+ * any additional dependencies outside the JDK. It might be useful when an application is using a
+ * relational database but something other than JDBI, for example Hibernate or another ORM or ORM-like
+ * framework such as jOOQ.
+ */
+@SuppressWarnings({ "SqlDialectInspection", "SqlNoDataSourceInspection" })
+public class JdbcApplicationErrorDao implements ApplicationErrorDao {
+
+    private static final int DEFAULT_PAGE_SIZE = 10;
+    private final DataSource dataSource;
+
+    public JdbcApplicationErrorDao(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public Optional<ApplicationError> getById(long id) {
+        try (var conn = connection();
+             var ps = conn.prepareStatement("select * from application_errors where id = ?")) {
+
+            ps.setLong(1, id);
+
+            try (var resultSet = ps.executeQuery()) {
+                if (resultSet.next()) {
+                    var applicationError = ApplicationErrorJdbc.mapFrom(resultSet);
+                    return Optional.of(applicationError);
+                }
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new ApplicationErrorJdbcException(e);
+        }
+    }
+
+    @Override
+    public long countResolvedErrors() {
+        return countUsingQuery("select count(id) from application_errors where resolved = true");
```

Comment:
We're using the literal `true` and `false` in queries which works for certain databases but not others (hi MySQL). 

Not necessarily now but might be a good idea to change all the literal booleans to bind parameters, both in JDBI and JDBC implementations.

Reply:
Shifting this to a discussion #320